datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -149,10 +149,6 @@ impl OptimizerRule for PushDownProjection {
             {
                 let mut used_columns: HashSet<Column> = HashSet::new();
                 if projection_is_empty {
-                    let field = find_small_field(scan.projected_schema.fields()).ok_or(
```

Comment:
I wonder if we can remove it as well for `LogicalPlan::Values`?

Reply:
yes, we can remove it.